src/rosdep2/main.py

```diff
@@ -476,6 +479,19 @@ def _package_args_handler(command, parser, options, args):
                 print('Skipping non-existent path ' + path)
         set_workspace_packages(ws_pkgs)
 
+        # Lookup package names from ament index.
+        if AMENT_PREFIX_PATH_ENV_VAR in os.environ:
+            if options.verbose:
+                print(
+                    'Searching ' + AMENT_PREFIX_PATH_ENV_VAR + ' for '
+                    'sources: ' + str(os.environ[AMENT_PREFIX_PATH_ENV_VAR].split(':')))
+            ws_pkgs = get_workspace_packages()
+            pkgs = get_packages_with_prefixes().keys()
+            ws_pkgs.extend(pkgs)
+            # Make packages list unique
+            ws_pkgs = list(set(ws_pkgs))
+            set_workspace_packages(ws_pkgs)
```

Comment:
> I don't think it matters if both paths set the same package name string

It is unlikely that both environment variables contain the same paths. I would actually expect them to be disjoint. So merging both results makes the most sense.

Reply:
I agree that merging them seems to make more sense. Even if there is overlap, it should be fine to merge them since we're just checking for existence of packages.